### ISSUE TITLE
game states and their handling/controls

### ISSUE THREAD

**Issue by hegemonkhan** [KVonGit/Quest-Group_Project#24]
for the main/starting ("base of operations", the main/starting hub) location/Object/Room, or as a separate Object (or using the 'game' Game Settings Object), we need to have the handling/controls for the game state.

Pixie can probably be of more help with this, and probably you're all aware of it as well too. As a 'game state' can be used to handle/control events/actions that effect the different worlds/time-periods and/or anything within them such as npc dialogue/conversation/topics/etc or various objects/items/etc.

Also, need to decide upon all of the different game states too, as well

basically just created this topic/thread, as while I know you're all aware of it, I think it is something we got to be thinking about first, as it effects the design of the rest of the game.

**Comment by rheadkid** [comment]
Is it pretty much like an attribute?

**Comment by hegemonkhan** [comment]
I'm not that knowledgeable, Pixie or others might be able to speak about this with more knowledge, but I think usually it's an Integer Attribute, for a stupid-simple example:

'portal' Object:

```
if (game.state = 0) {
  // move player to the first-easiest area of the game
} else if (game.state = 99) {
  // move player to the last-hardest area of the game
}
```

but, this is just an example, it'd be used for whatever differences in game state effects within our game (such as another example: npc topics, you can have a completely useless npc, and then when the 'game state' Integer Attribute is changed to its highest number/state, now this completely useless npc, has a topic on how to win the game/beat the final boss/whatever)

**Comment by hegemonkhan** [comment]
the implementation, as can be seen, is very simple: an Integer Attribute and the checking of it.

but, we still got to actually decide upon what happens with each of those states, which isn't so simple, as it involves our game design and the game-playing-progression within our game and etc etc etc.

for example, 'game.state = 0', could be a 'no turn' state, where we'd check for 'game.state=0', and if so, have the scripting for causing 'no turns' to occur while it remains: game.state = 0, though upon 'game.state' changing from '0' to some other value, we got scripting for causing the 'turns' to again be in effect.

This is not too difficult to implement (the special 'changed' Script and/or Turnscripts/Timers), but we got to plan this type of stuff for the various game states all out, of course.
